doc/managing-dependencies-in-cmake.md

```diff
@@ -0,0 +1,284 @@
+developers to create these files is outside the scope of the `google-cloud-cpp`
+project.
+
+In general, we will support four modes for Abseil:
+
+1. [`vcpkg`](https://github.com/Microsoft/vcpkg) is a Microsoft tool to install
+   many Open Source libraries on Windows.  It provides `unofficial::absl::`
+   packages for Abseil. We expect this to be popular on Windows.
+   When `GOOGLE_CLOUD_CPP_ABSEIL_PROVIDER` is set to `vcpkg` we will define
+   CMake [`INTERFACE`](https://cmake.org/cmake/help/v3.5/command/add_library.html?highlight=interface)
+   libraries with the `absl::*` names that automatically link the corresponding
+   `unofficial::absl::*` libraries.
+
+1. [`pkg-config`](https://www.freedesktop.org/wiki/Software/pkg-config/) is a
+   helper program on Unix to discover the compiler and linker command-line
+   arguments required to use a given library. When
+   `GOOGLE_CLOUD_CPP_ABSEIL_PROVIDER` is set to `pkg-config` we will define
+   `INTERFACE` libraries for each `absl::*` library used in `google-cloud-cpp`.
+   These `INTERFACE` libraries will set their target
+   [properties](https://cmake.org/cmake/help/v3.5/manual/cmake-properties.7.html#properties-on-targets)
+   based on the configuration flags discovered via `pkg-config`.
+   
+1. `package`: When `GOOGLE_CLOUD_CPP_ABSEIL_PROVIDER` is set to `package` we
+    will use `find_package(absl)` to find the Abseil libraries, and assume that
+    the system contains `Findabsl.cmake` files that tell CMake how to create the
+    `absl::*` targets.
+    
+1. `module`: When `GOOGLE_CLOUD_CPP_ABSEIL_PROVIDER` is set to `module`
+   (the default) we will simply add the `third_party/abseil` subdirectory to
+   the CMake build.
+
+### gRPC and Protobuf
+
+gRPC can be compiled using CMake, plain GNU Make, Bazel, or vcpkg.  Depending
+how it is compiled it may offer an install target (GNU Make and CMake have
+them), and it may or may not have installed config files for `find_package()`.
+
+Like with Abseil we will support four different configurations for gRPC:
+
+1. [`vcpkg`](https://github.com/Microsoft/vcpkg) packages gRPC, and provides
+   targets (`gRPC::gprc++` and `gRPC::grpc`) for CMake.  
+   When `GOOGLE_CLOUD_CPP_GRPC_PROVIDER` is set to `vcpkg` we will enable
+   these targets using `find_package()`.  We will also use `find_package()` to
+   find the `protobuf` library.  The package introduces a
+   `protobuf::libprotobuf` target.
+
+1. [`pkg-config`](https://www.freedesktop.org/wiki/Software/pkg-config/) when
+   compiled and installed with GNU Make gRPC installs `pkg-config` support
+   files. When `GOOGLE_CLOUD_CPP_GRPC_PROVIDER` is set to `pkg-config` we will
+   define `INTERFACE` libraries for `gRPC::grpc++` and `gRPC::grpc`each `absl::*`
+   library used in `google-cloud-cpp`.  These `INTERFACE` libraries will set
+   their target [properties](https://cmake.org/cmake/help/v3.5/manual/cmake-properties.7.html#properties-on-targets)
```

Comment:
URL Can be moved to new line as not fit in a single line.

Reply:
Use markdown reference links to make this more readable.